Add ExitOnInterrupt utils function; catch interrupts when using wait

caesarshift opened the pull request:
Add ExitOnInterrupt utils function; catch interrupts when using wait

## Description

Add ability to catch interrupts when using wait command.

## Related Issue

Fixes #1864

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

caesarshift commented:
The ticket (and this PR) is specific to the `wait` command, but this same approach could be applied to other commands as well (e.g., `zarf connect registry` has the same issue)

Racer159 commented on src/cmd/tools/wait.go:
```diff
@@ -29,6 +30,7 @@ var waitForCmd = &cobra.Command{
 	Example: lang.CmdToolsWaitForExample,
 	Args:    cobra.MinimumNArgs(2),
 	Run: func(cmd *cobra.Command, args []string) {
+		exec.ExitOnInterrupt(1, lang.ErrInterrupt)
```
Can we make this call in the `rootCmd` `PersistentPreRun` to handle all exits for Zarf commands (after the check for vendored commands)?

https://github.com/defenseunicorns/zarf/blob/4d4ae8fe077b03ef06f2674cdcb5d310c04bb125/src/cmd/root.go#L39

That would allow it to work for this command and be a good starting point for https://github.com/defenseunicorns/zarf/issues/1518

Racer159 commented on src/pkg/utils/exec/exec.go:
```diff
@@ -207,3 +209,17 @@ func GetOSShell(shellPref types.ZarfComponentActionShell) (string, string) {
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		if ErrorMessage != "" {
+			message.Warnf(ErrorMessage)
+		}
+		os.Exit(ErrorCode)
```
```suggestion
		message.Fatal(lang.ErrInterrupt, lang.ErrInterrupt.Error())
```

This might be simpler for now and would show be more consistent with other interrupts (like when a confirmation is canceled with an interrupt).

This changes Warnf to Fatal to print a red error rather than a warning and remove formatting (since that can cause garbled text to show if the text contains formatting shortcodes).  It also changes `lang.ErrInterrupt` to be a reusable error rather than a message (see the bottom of `lang/english.go`)

![image](https://github.com/defenseunicorns/zarf/assets/3977569/8033153c-6416-427b-b20f-a5c07bcddc76)

Racer159 commented on src/pkg/utils/exec/exec.go:
```diff
@@ -207,3 +209,17 @@ func GetOSShell(shellPref types.ZarfComponentActionShell) (string, string) {
 }
+
+// ExitOnInterrupt catches an interrupt and exits with ErrorCode
+func ExitOnInterrupt(ErrorCode int, ErrorMessage string) chan os.Signal {
```
```suggestion
func ExitOnInterrupt() chan os.Signal {
```
Per the below this likely can be simplified to just use a reusable error

Racer159 commented on src/config/lang/english.go:
```diff
@@ -27,6 +27,7 @@ const (
 	ErrCreatingDir         = "failed to create directory %s: %s"
 	ErrRemoveFile          = "failed to remove file %s: %s"
 	ErrUnarchive           = "failed to unarchive %s: %s"
+	ErrInterrupt           = "Failed due to interrupt"
```
Per the below this may be better as a reusable error.

Racer159 reviewed (changes requested):
Thanks for the PR! Just a few small change but otherwise looks good.

caesarshift commented:
Thank you for the feedback. Beyond your changes, I've pushed the following additional changes:
- Minor cleanup to the `ExitOnInterrupt` function: removed unused return variable; changed size of channel to 1
- Added the `exec.ExitOnInterrupt` to the `PersistentPreRun` in `rootCmd`, `toolsCmd`, and `registryCmd`. Adding it to the `toolsCmd` and `registryCmd` was required because they define their own `PersistentPreRun` functions that override the rootCmd's function.
